rbd/migration_test.go

```diff
@@ -0,0 +1,164 @@
+//go:build !(octopus || nautilus) && ceph_preview
+// +build !octopus,!nautilus,ceph_preview
+
+package rbd
+
+import (
+	"testing"
+	"time"
+
+	"github.com/ceph/go-ceph/rados"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func createAndWriteDataToImage(t *testing.T, ioctx *rados.IOContext) string {
+	opts := NewRbdImageOptions()
+	defer opts.Destroy()
+
+	assert.NoError(t,
+		opts.SetUint64(ImageOptionOrder, uint64(testImageOrder)))
+
+	name := GetUUID()
+	err := CreateImage(ioctx, name, 1<<25, opts)
+	require.NoError(t, err)
+
+	img, err := OpenImage(ioctx, name, NoSnapshot)
+	require.NoError(t, err)
+
+	img.CreateSnapshot("snap1")
+
+	_, err = img.WriteAt([]byte("sometimes you feel like a nut"), 0)
+	assert.NoError(t, err)
+
+	err = img.Close()
+	assert.NoError(t, err)
+
+	return name
+}
+
+func TestMigration(t *testing.T) {
+	conn := radosConnect(t)
+
+	pool := GetUUID()
+
+	err := conn.MakePool(pool)
+	require.NoError(t, err)
+
+	ioctx, err := conn.OpenIOContext(pool)
+	require.NoError(t, err)
+
+	name := createAndWriteDataToImage(t, ioctx)
+	destImage := GetUUID()
+
+	err = MigrationPrepare(ioctx, ioctx, name, destImage, NewRbdImageOptions())
+	require.NoError(t, err)
+
+	var status *MigrationImageInfo
+	require.Eventually(t, func() bool {
+		status, err = MigrationStatus(ioctx, destImage)
+		return err == nil
+	}, 5*time.Second, 10*time.Millisecond)
```

Comment:
I don't think you need this `require.Eventually` thing. It should succeed on the first call. And the same for all other cases below.

Reply:
Agreed, please try removing these and if the runs become unstable without them we'll just need ton investigate further.

Reply:
Made the change.